optimiser/optimisation_problem.py

```diff
@@ -0,0 +1,220 @@
+        )
+
+        # Rate of electricity entering the battery cannot exceed the maximum charge rate
+        problem += (
+            electricity_to_battery[t] <= MAX_BATTERY_CHARGE_RATE * TIMESLICE_SIZE,
+            f"Battery charge rate {t}",
+        )
+
+        # The battery cannot exceed its maximum capacity
+        problem += (
+            battery_state_of_charge[t] <= MAX_BATTERY_CAP,
+            f"Battery capacity {t}",
+        )
+
+        # The battery cannot be less than 0
+        problem += (
+            battery_state_of_charge[t] >= 0,
+            f"Battery minimum state of charge {t}",
+        )
+
+        # Financial constraints
+        problem += (
+            (grid_electricity_to_house[t] + grid_electricity_to_battery[t])
+            * UNIT_PRICE_GRID_ELECTRICITY
+        ) - grid_electricity_costs[t] == 0, f"Grid electricity costs {t}"
+        problem += (
+            battery_electricity_to_house[t] * UNIT_PRICE_BATTERY_ELECTRICITY
+        ) - battery_electricity_costs[t] == 0, f"Battery electricity costs {t}"
+        problem += (
+            renewable_electricity_to_house[t] * UNIT_PRICE_RENEWABLE_ELECTRICITY
+        ) - renewable_eletricity_costs[t] == 0, f"Renewable electricity costs {t}"
+        problem += (
+            battery_to_grid_sales[t] * UNIT_PRICE_BATTERY_SALE_PRICE
+        ) - battery_to_grid_sales[t] == 0, f"Battery to grid sales {t}"
+
+    problem.writeLP("BatteryOptimisation.lp")
+    problem.solve()
+    print(pl.LpStatus[problem.status])
+
+    print("---- Costs ----")
+    print("Total Cost = £", pl.value(problem.objective))
+    print(
+        "Total Battery Costs = ",
+        sum([pl.value(battery_electricity_costs[t]) for t in T]),
+    )
+    print("Total Grid Costs = ", sum([pl.value(grid_electricity_costs[t]) for t in T]))
+    print(
+        "Total Renewable Costs = ",
+        sum([pl.value(renewable_eletricity_costs[t]) for t in T]),
+    )
+
+    print("---- Energy Flows ----")
+    print(
+        f"Total Electricity Sent from Battery to Grid = {sum([pl.value(battery_to_grid_sales[t]) for t in T])} kWh"
+    )
+    print(
+        f"Total Renewable Generation = {sum([pl.value(total_renewable_generation[t]) for t in T])} kWh"
+    )
+    print(
+        f"Total Electricity Sent from Renewables to House = {sum([pl.value(renewable_electricity_to_house[t]) for t in T])} kWh"
+    )
+    print(
+        f"Total Electricity Sent from Renewables to Battery = {sum([pl.value(renewable_electricity_to_battery[t]) for t in T])} kWh"
+    )
+    print(
+        f"Total Electricity Sent from Battery to House = {sum([pl.value(battery_electricity_to_house[t]) for t in T])} kWh"
+    )
+    print(
+        f"Total Electricity Sent from Grid to House = {sum([pl.value(grid_electricity_to_house[t]) for t in T])} kWh"
+    )
+
+    print("---- Battery ----")
+    print(
+        f"Total Battery Degradation = {sum([pl.value(battery_degradation[t]) for t in T])} kWh"
+    )
```

Comment:
Move the post-processing into a separate module - again raise in another issue ticket.

Reply:
Raised in ticket: https://github.com/davidwickh/Home-Battery-Optimiser/issues/7